morango/models.py

```diff
@@ -100,10 +96,33 @@ class AbstractStoreModel(models.Model):
     last_saved_instance = models.UUIDField()
     last_saved_counter = models.IntegerField()
     last_saved_counter_per_instance = models.TextField(default="{}")  # RMC
+    model_name = models.CharField(max_length=40)
+
+    class Meta:
+        abstract = True
+
+
+class AbstractDatabaseMaxCounter(models.Model):
+
+    instance_id = models.UUIDField()
+    max_counter = models.IntegerField()
+    transfer_session_id = UUIDField()
+    kind = models.BooleanField()
 
     class Meta:
         abstract = True
 
+    @classmethod
+    def filter_max_counters(cls, query):
+        query = json.loads(query)
+        filters = []
+        for key, value in six.iteritems(query):
+            filters.append(Q(**{key: value}) | Q(**{key: "*"}))
+
+            query = reduce(lambda x, y: x & y, filters)
+            rows = cls.objects.filter(query)
+            return rows.values('instance_id').annotate(max_counter=Max('max_counter'))
```

Comment:
I'm pretty sure its annotate. I wrote some preliminary tests for this and it seemed to be working fine. I can push those tests in a separate PR.